### ISSUE TITLE
When creating a file, the console reports an error. The error message is as follows

### ISSUE THREAD

**Issue by bj-white** [theia-ide/theia#4403]
messages.js:46 Uncaught (in promise) Error: Request createFile failed with message: EACCES: permission denied, open '/home/project/Untitled.txt'
    at Proxy.<anonymous> (proxy-factory.ts:232)
    at workspace-commands.ts:181

**Comment by kittaakos** [comment]
From https://github.com/theia-ide/theia/issues/4402#issue-413284179:

> logger-protocol.ts:112 root ERROR Error: Request 'setContent' failed
at Proxy. (http://192.168.153.138:3000/bundle.js:52:133973)
at e. (http://192.168.153.138:3000/bundle.js:30:148557)
at http://192.168.153.138:3000/bundle.js:30:146446
at Object.next (http://192.168.153.138:3000/bundle.js:30:146551)
at a (http://192.168.153.138:3000/bundle.js:30:145297)

**Comment by kittaakos** [comment]
@bj-white, you do not have write access in your Theia workspace. What are you trying to do?

**Comment by bj-white** [comment]
Hello, how can I write permissions?

**Comment by kittaakos** [comment]
You need to open a workspace inside a folder where the current OS user has write access. What did you do?

**Comment by opiation** [comment]
I am getting the same issues.  I run the following, as perscribed by the home page:

```shell
docker run --publish 3000:3000 --volume ${PWD}/home:/home/project theiaide/theia
```

Docker logs reveal the following before even opening `localhost:3000`:

```
Server worker has been started. [ID: 1 | PID: 11]
Error: @postConstruct error in class TypeScriptContribution: EACCES: permission denied, open '/home/theia/.application-module-loader.js'
at _postConstruct (/home/theia/node_modules/inversify/lib/resolution/instantiation.js:30:19)
at Object.resolveInstance (/home/theia/node_modules/inversify/lib/resolution/instantiation.js:47:5)
at /home/theia/node_modules/inversify/lib/resolution/resolver.js:72:42
at /home/theia/node_modules/inversify/lib/resolution/resolver.js:34:24
at Array.map (<anonymous>)
at /home/theia/node_modules/inversify/lib/resolution/resolver.js:32:34
at Object.resolve (/home/theia/node_modules/inversify/lib/resolution/resolver.js:96:12)
at /home/theia/node_modules/inversify/lib/container/container.js:319:37
at Container._get (/home/theia/node_modules/inversify/lib/container/container.js:310:44)
at Container.getAll (/home/theia/node_modules/inversify/lib/container/container.js:239:21)
```

Inspecting the running processes reveals all processes are started by user `theia` (UID 1001)
```
$ ps
PID   USER     TIME  COMMAND
    1 theia     0:00 node /home/theia/src-gen/backend/main.js /home/project --hostname=0.0.0.0
   11 theia     0:01 /usr/local/bin/node /home/theia/src-gen/backend/main.js /home/project --hostname=0.0.0.0
   17 theia     0:03 /usr/local/bin/node /home/theia/node_modules/@theia/core/lib/node/messaging/ipc-bootstrap.js
   25 theia     0:00 /bin/bash
   42 theia     0:03 /usr/local/bin/node /home/theia/node_modules/@theia/core/lib/node/messaging/ipc-bootstrap.js
   51 theia     0:00 /bin/bash
   91 theia     0:00 bash
  102 theia     0:00 ps
```

```
$ id theia
uid=1001(theia) gid=1001(theia) groups=1001(theia),1001(theia)
```

Unfortunately, nothing in `/home` is owned by user `theia`.  `/home/node` and `/home/project` are owned by user `node` with UID 1000 and `/home/theia` is owned by an unknown user with UID 2001.

```
$ ls -la /home
drwxr-sr-x    2 node     node          4096 Mar  8 00:48 node
drwxrwxrwx    2 node     node          4096 Mar 22 23:11 project
drwxr-sr-x    1 2001     2001          4096 Mar 22 00:23 theia
```

For what it's worth, `${PWD}/home` is owned by my docker host user that's runnning the image with UID 1000 (on the docker host).  How is this intended to work?  It seems like the user permissions within `/home` are not properly set up.  I was able to make it work by using docker as root and changing file permissions:

```shell
docker exec --user root theia chown -rc 1001:1001 /home/theia
docker exec --user root theia chown -rc 1001:1001 /home/project
```

But then the mounted `/home/project` is inaccessible from my docker host user.

**Comment by akosyakov** [comment]
@opiation I cannot reproduce it :( Do you see something suspicious here: https://github.com/theia-ide/theia-apps/blob/cddc5ea11e5438335098f58dbec5f7794e7c0d8b/theia-docker/Dockerfile#L20-L25 It should grant access rights to `/home/theia` and `/home/projct` for theia user.

**Comment by opiation** [comment]
Apologies.  I had missed a detail which became apparent after retrying on a different machine.  I only encountered these errors when running the image as prescribed but with the docker daemon being user-namespaced.  I did not encounter these errors when I disabled the user-namespacing and retried with the exact same commands.

I suppose since the example instruction on your site does not mention user namespacing, it technically works as _intended_.  Nonetheless, since user-namespacing docker is quite useful and commonplace when securing uses of docker in production, it may be useful to prospective `theia` users to advise them now that the example does not currently work as is with docker being user-namespaced.

My experience with docker has been that, in general, docker images are intended to function similarly with or without user namespacing.  Depending on what you think of this expectation or how closely @bj-white thinks this aligns with their report, you may consider this a bug.  If anyone would prefer this be a separate bug report or feature request, let me know.

For the record, I'm uncertain what change to the Dockerfile would enable this image to work identically in both circumstances.  I have little experience with Unix permissions and namespacing in this sort of usage so I can't really be more helpful at this time.

**Comment by blastbeng** [comment]
@opiation 
I have the same exact error, i've seen you've talked abot user namespacing

How did you solve this error?

I have a folder owned by root user and trying to run theia as root on that folder, but getting permission denied when saving\creating\deleting new files

I've checked inside theia that it is running as theia user

**Comment by piec** [comment]
> `--user youruser`

It seems to work but it's not great because this will make the theia processes unable to write to `/home/theia`. And apparently it needs to in order to work properly
Example:
```
$ docker run --name theia -it -p 3000:3000 --user 1000 -v "$GOPATH:/home/project:cached" theiaide/theia:next
Error: @postConstruct error in class TypeScriptContribution: EACCES: permission denied, open '/home/theia/.application-module-loader.js'                                                                                                                                                                                    
    at _postConstruct (/home/theia/node_modules/inversify/lib/resolution/instantiation.js:30:19)                                                                                                                                                                                                                            
    at Object.resolveInstance (/home/theia/node_modules/inversify/lib/resolution/instantiation.js:47:5)                                                                                                                                                                                                                     
[...]
```
---
A quick solution would be to make /home/theia rwX to `o` at build and run with theia process with the host's user uid.

**Comment by kmpm** [comment]
I have the same issue with a default installed "docker.io" on ubuntu 18.04.
When I run with a mapped volume for /home/project, that folder is owned by the same uid:gid as the host user. 
If I don't map a volume it's is owned by theia:theia as @akosyakov rightly mentions in an earlier comment.

**Comment by perguth** [comment]
Related issue, also because of the `1001` uid: https://github.com/eclipse-theia/theia/issues/3198

**Comment by perguth** [comment]
I got it to work adding `-u $(id -u ${USER}):$(id -g ${USER})` to the docker command given in the [README](https://github.com/theia-ide/theia-apps#theia-docker) (see: https://jtreminio.com/blog/running-docker-containers-as-current-host-user/).

👋 **So overall it should be:** `docker run -it --init -p 8080:3000 -u $(id -u ${USER}):$(id -g ${USER}) -v "$(pwd):/home/project:cached" theiaide/theia:next`

**Comment by GavinRay97** [comment]
@perguth Thank you!

**Comment by frankgerhardt** [comment]
Happy anniversary, this issue is one year old. I just ran `docker run -it --init -p 3000:3000 -v "$(pwd):/home/project:cached" theiaide/theia:next` and I could not even create a file in `/home/project/Untitled.txt` beause of a permission problem!?

```
root ERROR Request createFile failed with error: EACCES: permission denied, open '/home/project/Untitled.txt' Error: EACCES: permission denied, open '/home/project/Untitled.txt'
```

**Comment by vrkansagara** [comment]
> I got it to work adding `-u $(id -u ${USER}):$(id -g ${USER})` to the docker command given in the [README](https://github.com/theia-ide/theia-apps#theia-docker) (commands explained here: https://jtreminio.com/blog/running-docker-containers-as-current-host-user/).
> 
>  **Tldr:** `docker run -it --init -p 3000:3000 -u $(id -u ${USER}):$(id -g ${USER}) -v "$(pwd):/home/project:cached" theiaide/theia:next`

working for me.

**Comment by perguth** [comment]
Somebody may donate time or money for this? 🧡

**Comment by TEFEdotCC** [comment]
Please fix the problem, the solution of @frankgerhardt works for me too.

**Comment by ezzio-salas** [comment]
> I got it to work adding `-u $(id -u ${USER}):$(id -g ${USER})` to the docker command given in the [README](https://github.com/theia-ide/theia-apps#theia-docker) (commands explained here: https://jtreminio.com/blog/running-docker-containers-as-current-host-user/).
> 
> 👋 **Tldr:** `docker run -it --init -p 3000:3000 -u $(id -u ${USER}):$(id -g ${USER}) -v "$(pwd):/home/project:cached" theiaide/theia:next`

This worked for me fixing my permissions issues.
Thanks **vrkansagara**

**Comment by FreezeSoul** [comment]
@perguth it works for me , thk

**Comment by sgraband** [comment]
Stale 2019 Docker permission question (EACCES).
